walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,117 @@
 // limitations under the License.
 
 package com.google.sps;
-
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // throw new UnsupportedOperationException("TODO: Implement this method.");
+    HashSet<String> attendees = new HashSet<String>(request.getAttendees());
+    HashSet<String> optionalAttendees = new HashSet<String>(request.getOptionalAttendees());
+    ArrayList<TimeRange> badTimes = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> optionalBad = new ArrayList<TimeRange>();
+    Collection<TimeRange> range = new ArrayList<TimeRange>();
+    Collection<TimeRange> OptRange = new ArrayList<TimeRange>();
+    long meetingDuration = request.getDuration();
+    int duration = (int) meetingDuration;
+
+    //Base Cases
+    if(request.getAttendees().size() < 1){
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    if(meetingDuration > 1440){  // 1440 is the timeRange whole day in minutes
+        return Arrays.asList();
+    }
+
+    if(events.size()<1){
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    //Put all of the bad times in list   : if it is related to an attendee
+    for (Event event : events) {
+        int eventStart = event.getWhen().start();
+        int eventDuration = event.getWhen().duration();
+        Iterator<String> i = event.getAttendees().iterator();  
+        while(i.hasNext()) {  
+           String a = i.next();
+           if(attendees.contains(a)){
+            badTimes.add(TimeRange.fromStartDuration(eventStart,eventDuration));
+           }
+           else if(optionalAttendees.contains(a)){
+            optionalBad.add(TimeRange.fromStartDuration(eventStart,eventDuration));
+           }
+        }  
+    }
+    
+    if(badTimes.size()<1){
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    // Sort unavailable times list
+    Collections.sort(badTimes, TimeRange.ORDER_BY_START);
+    ArrayList<TimeRange> finalUnavailable = new ArrayList<TimeRange>();
+    boolean lastAdded = false;
+   
+    //Combine unavailable times if they overlap
```

Comment:
What would happen if there are multiple overlaps? Would this actually merge them all into one? For example:

_| ...................... |
____| ...................... |
_______| .......... |